src/error.rs

```diff
@@ -7,7 +7,11 @@ pub enum EvalError {
     DanglingPointerDeref,
     InvalidBool,
     InvalidDiscriminant,
-    PointerOutOfBounds,
+    PointerOutOfBounds {
+        offset: usize,
+        size: usize,
+        len: usize,
```

Comment:
Can you rename `len` to something like `alloc_size` or `alloc_len` to make this clearer? It might be worth sticking the `AllocId` in here, too.

(Really all the error variants need more information in them, which I have neglected so far.)